pynxtools_xps/reader_utils.py

```diff
@@ -298,12 +298,13 @@ def construct_detector_data_key(spectrum):
 
 def construct_entry_name(key):
     """Construction entry name."""
-    key_parts = key.split("/")
-    try:
-        # entry example : sample__name_of_scan_region
-        entry_name = (
-            f'{key_parts[2].split("_", 1)[1]}' f"__" f'{key_parts[4].split("_", 1)[1]}'
-        )
-    except IndexError:
-        entry_name = ""
-    return entry_name
+    name_parts = []
+
+    for key_part in ["Group_", "Region_", "RegionData_"]:
+        try:
+            pattern = rf"{key_part}(.*?)(?=\/|$)"
+            name_parts += [re.search(pattern, key).group(1)]
```

Comment:
Consider using `re.compile`. Maybe even outside the function if this is run many times.

Reply:
done